codec.go

```diff
@@ -165,6 +166,36 @@ func addressStringToBytes(p Protocol, s string) ([]byte, error) {
 		binary.BigEndian.PutUint16(b, uint16(i))
 		return b, nil
 
+	case P_ONION:
+		addr := strings.Split(s, ":")
+		if len(addr) != 2 {
+			return nil, fmt.Errorf("failed to parse %s addr: %s does not contain a port number.", p.Name, s)
```

Comment:
well... perhaps. however i'd argue that an onion address without a port number is useless. an onion address could have several ports listening and each port could correspond to a different service hosted on the same tor instance.